### PR TITLE
Update LLB.convert_caresian_to_polar()

### PR DESCRIPTION
**Pull request by Nilodirf**
Completed method to convert from cartesian back to polar coordinates.

### CODE REVIEW

**Comment by Nilodirf** [comment]
missing still ill-defined illdefined phi for r=0. For x=y=0, phi=0 and gamma=0 is returned.

File: udkm1Dsim/simulations/magnetization.py [hunk 1]
```diff
@@ -1013,6 +1013,18 @@ def convert_cartesian_to_polar(cartesian):
...
+        xs=cartesian[..., 0]
+        ys=cartesian[..., 1]
+        zs=cartesian[..., ]
+        trans_amplitudes=np.hypot(xs, ys)
+        amplitudes=np.hypot(trans_amplitudes, zs)
+        zeromag_mask=amplitudes!=0.
+        polar[..., 0]=amplitudes   
+        polar[..., 1]=np.arctan2(ys, xs)               
+        polar[zeromag_mask, 2]=np.arccos(np.divide(zs,amplitudes))
+        polar[~zeromag_mask, 2]=0.
```
Comment (dschick) [inline udkm1Dsim/simulations/magnetization.py]: you do not need that line, since the `polar` array is initialized as zeros

**Comment by dschick** [comment]
Hi @Nilodirf ,

Thanks for adding this code. Could you please also add some docstring to the method, ideally with the math?

Moreover, please install and run `flake8` on the file you gave changed, so keep a consistent code style.

Best

Daniel

File: udkm1Dsim/simulations/magnetization.py (hunk 1 above)
Comment (Nilodirf) [reply udkm1Dsim/simulations/magnetization.py]: Codeline was cut and checked with flake8 before the last commit

**Comment by dschick** [comment]
after merging and polishing, I noticed, that the conversion from cartesian to polar and back to caresian, does not result in the initial coordinates.

Could you please have a look for that?

**Pull request merged**